### DISCUSSION TITLE
Budibase Theming

### DISCUSSION THREAD

**Discussion by shogunpurple** [Budibase/budibase#772] [Ideas]
As a user, it is beneficial to be able to define some variables/themes at the top level of an application that can be used to control the look and feel of your application by changing a single color or applying a theme.

In budibase, styling the whole app in this way is difficult at the moment. My suggestion is that we define a "spec" for components that make use of some Budibase CSS variables that can be set at the top level:
- `--primary-color`
- `--secondary-color`
- `--tertiary-color`

All budibase standard components will know about these - and they can be leveraged by component authors also, as well as adding their own.

**Comment by PGLange** [comment]
Hey @ll,
did you schedule when the app styling will be useable? I really need to "style" the header, and I would appreciate an css upload area, where a css / scss could be uploaded to style components all over the app :)

**Reply by joebudi** [reply]
Hey @PGLange 

What styles do you need to apply to the header?

**Reply by GuilhermeHirsch** [reply]
It would be interesting to have the possibility to put icons in the menu options, as it is a standard in applications. Will it be possible to apply dark mode in the menu, @joebudi?

**Reply by joebudi** [reply]
This is a cool idea but it's not a blocker so it won't come in the next release.  The nav will accept a dark mode theme.
